testing/test_unittest.py

```diff
@@ -735,3 +735,23 @@ def test_skip(self):
         *SKIP*[1]*skipping due to reasons*
         *1 skipped*
     """)
+
+def test_class_method_containing_test_issue1558(testdir):
+    testdir.makepyfile(test_foo="""
+        import unittest
+
+        class MyTestMeta(type):
```

Comment:
Oh right, sorry. Please set the attribute inside the class (into the function object) instead.

```python
import unittest

class MyTestCase(unittest.TestCase):

    def test_should_run(self):
        pass
    def test_should_not_run(self):
        pass
    test_should_not_run.__test__ = False
```